Multisig/Features/Security/SensitiveEncryptedStore.swift

```diff
@@ -0,0 +1,183 @@
+    func find(address: Address, password: String? = nil) throws -> EthPrivateKey? {
+        // find encrypted private key for the address
+        guard let encryptedPrivateKeyData = try? keychainStorage.retrieveEncryptedData(account: address.checksummed) else {
+            return nil
+        }
+
+        // find sensitiveKEK
+        let s = password != nil ? password : keychainStorage.retrievePasscode()
+        let sensitiveKEK = try keychainStorage.findKey(query: SQuery.ecPrivateKey(tag: KeychainStorage.sensitiveKekTag, password: s?.data(using: .utf8)))!
+
+        // find encrypted sensitive key
+        let encryptedSensitiveKeyData = try keychainStorage.retrieveEncryptedData(account: KeychainStorage.sensitiveEncryptedPrivateKeyTag)!
+
+        // decrypt encrypted sensitive key
+        var error: Unmanaged<CFError>?
+        guard let decryptedSensitiveKeyData = SecKeyCreateDecryptedData(sensitiveKEK, .eciesEncryptionStandardX963SHA256AESGCM, encryptedSensitiveKeyData as CFData, &error) else {
+            throw error!.takeRetainedValue() as Error
+        }
+        // Data -> key
+        LogService.shared.debug("decryptedSensitiveKeyData: \((decryptedSensitiveKeyData as Data).toHexString())")
+        let attributes: NSDictionary = [
+            kSecAttrKeyType: kSecAttrKeyTypeECSECPrimeRandom,
+            kSecAttrKeySizeInBits: 256,
+            kSecAttrKeyClass: kSecAttrKeyClassPrivate
+        ]
+        guard let decryptedSensitiveKey: SecKey = SecKeyCreateWithData(decryptedSensitiveKeyData, attributes, &error) else {
+            // will fail here if password was wrong
+            throw error!.takeRetainedValue() as Error
+        }
+        LogService.shared.debug("decryptedSensitiveKey: \(decryptedSensitiveKey)")
+        // decrypt the eth key with sensitive key
+        let decryptedEthKeyData = SecKeyCreateDecryptedData(decryptedSensitiveKey, .eciesEncryptionStandardX963SHA256AESGCM, encryptedPrivateKeyData as CFData, &error) as? Data
+
+        // Data -> String key
+        let decryptedEthKey = decryptedEthKeyData!.toHexString()
+        // return private key
+        return decryptedEthKey
+    }
+
+    func changePassword(from oldPassword: String, to newPassword: String) {
+        // create a new kek
+        // decrypt private senstivie key with kek
+        // encrypt sensitive key with new kek
+
+        // decrypt data key with old data kek
+        // encrypt data key with new data kek
+    }
+
+    func changeSettings() {
+        // Settings
+
+        //  change password
+        //  enable / disable password
+
+        //  use biometry
+        //  use password in addition to face id when signing
+
+    }
+
+    func delete(address: Address) {
+        // delete encrypted blob by address
+    }
+
+    // Copied from AuthenticationController
+    private func derivePasscode(from plaintext: String, useOldSalt: Bool = false) -> String {
+        let salt = "Safe Multisig Passcode Salt"
```

Comment:
I deleted `derivePasscode()` because we're not using it. When we derive the user given asscode, we might be able to just use the original implementation. Which takes care of the old salt.